### PR TITLE
Allow for `DateTime` clocks

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Should be useful for LESbrary simulations whose states might actually have an associated date and time from the real world.

But I'm not sure if we should merge this PR as `DateTime` from `Base.Dates` only has millisecond precision so you could get small errors in actual `model.clock.time` after many iterations (LESbrary simulations will probably run for millions of iterations).

Two small packages that might help us here
https://github.com/JeffreySarnoff/TimesDates.jl
https://github.com/FugroRoames/Chrono.jl

### CODE REVIEW

**Comment by glwagner** [comment]
I agree that something like this would be pretty convenient.

**Comment by ali-ramadhan** [comment]
So I modified this PR to support more general `AbstractTime` clocks. So now clocks work if you pass in a `DateTime` from Base.Dates (with millisecond precision) but also if you pass in a `TimeDate` from TimesDates.jl (with nanosecond precision) as they're both subtypes of `Dates.AbstractTime`.

Oceananigans does not need to depend on TimesDates.jl but I added it as a test dependency so we can make sure `TimeDate` always works as I think it will be useful for LESbrary simulations.

File: src/Models/clock.jl [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+
 Base.show(io::IO, c::Clock{FT}) where FT =
     println(io, "Clock{$FT}: time = ", prettytime(c.time), ", iteration = ", c.iteration)
+
+function increment_clock!(clock, Δt)
+    clock.time += Δt
+    clock.iteration += 1
+    return nothing
+end
+
```
Comment (glwagner) [inline src/Models/clock.jl]: could also call this “tick!” But looks good

Comment (ali-ramadhan) [reply src/Models/clock.jl]: Love it!

**Pull request merged**